src/xrGame/xrgame_dll_detach.cpp

```diff
@@ -61,8 +61,7 @@ extern CUIXml* pWpnScopeXml;
 
 void clean_game_globals()
 {
-    // destroy object factory
-    xr_delete(g_object_factory);
+    ai().DestroyScriptEngine();
```

Comment:
@FreeZoneMods, so, if we are deleting `g_object_factory`, then renderer scripts (`*.s` files) are trying to call destroyed factory...